sqlalchemy_bigquery/base.py

```diff
@@ -547,6 +547,13 @@ def visit_bindparam(
         bq_type = self.dialect.type_compiler.process(type_)
         bq_type = self.__remove_type_parameter(bq_type)
 
+        if bq_type == "JSON":
+            # FIXME: JSON is not a member of `SqlParameterScalarTypes` in the DBAPI
+            # For now, we hack around this by:
+            # - Rewriting the bindparam type to STRING
+            # - Applying a bind expression that converts the parameter back to JSON
+            bq_type = "STRING"
```

Comment:
**note** Without this, I think generated queries will crash when submitted to the DBAPI provided in `python-bigquery` because JSON is not a valid scalar type. SQLAlchemy generates query parameters in many contexts (via bindparams), so I do think this is necessary in the initial pass.

One alternative I considered was to unconditionally return "STRING" from `visit_JSON` (i.e: rewrite the type to STRING in all contexts) but this breaks DDL.

Reply:
It also looks like there's a pattern for rewriting the type based on the kind of target expression in the `visit` hook:

```python
        if isinstance(
            kw.get("type_expression"), Column
        ):  # column def
            ...
```

I will try this out, feels a bit cleaner / more consistent with what we have already

Reply:
Resolved in 65d676d